src/ethereum/networkAdapters/EvmScanAdapter.ts

```diff
@@ -349,7 +349,14 @@ export class EvmScanAdapter extends NetworkAdapter<EvmScanAdapterConfig> {
       tx.from.toLowerCase() ===
       this.ethEngine.walletLocalData.publicKey.toLowerCase()
     const tokenTx = currencyCode !== this.ethEngine.currencyInfo.currencyCode
-
+    const knownTokenId = Object.keys(this.ethEngine.allTokensMap).find(
+      tokenId =>
+        this.ethEngine.allTokensMap[tokenId].currencyCode === currencyCode
+    )
+    if (tokenTx && knownTokenId === undefined) {
+      throw new Error('Unknown token')
+    }
```

Comment:
This isn't really safe, because of the general currencyCode confusion. However, I see that we would need to change `GetTxsParams` from currencyCode to tokenId, which is a pretty deep refactor.

If will approve this for now, IF you create a follow-on task to actually make this change to `GetTxsParams`.

Reply:
https://app.asana.com/0/0/1206891509258292/f